tests/ashley/test_forum_search.py

```diff
@@ -399,5 +400,33 @@ def test_forum_search_restrict_poster_topic_forums(self):
         self.assertContains(response, post1.topic.slug)
         self.assertContains(response, post5.topic.slug)
 
+    def test_forum_search_empty_public_username(self):
```

Comment:
I'm wondering whether it's a good thing to index "anonymous" as the user's name.... Shouldn't we index a null value in this case?

Reply:
I was wondering the same thing :)

The thing is that we have to use the `get_forum_member_display_name()` shortcut in the `PostIndex` definition to get the username, because its behavior is customizable with the  `USER_DISPLAY_NAME_METHOD` machina setting.

The «Anonymous» default value is specific to our own implementation of this method and I think we should not make an exception to handle our use case here. I see 3 solutions here :

- Keep it like this and index the «Anonymous» username. 
- Add a new setting and shortcut function to get the username to index (with a default value that calls get_forum_member_display_name() )
- update our `get_public_username()` method on our User model to return a null value. But it implies to update the template helper function to handle the default «Anonymous» value.

WDYT ?

Reply:
I would say 2 or 3 but I can't tell which one is better from where I stand :stuck_out_tongue:

Reply:
Ok :) I'll do that in another PR then.